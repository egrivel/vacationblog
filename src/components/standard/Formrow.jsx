import React from 'react';
import PropTypes from 'prop-types';

class Formrow extends React.Component {
  static propTypes = {
    label: PropTypes.string,
    labelFor: PropTypes.string,
    children: PropTypes.object
  };

  render() {
    const {label, labelFor, children} = this.props;

    return (
      <div>
        <div
          className="formLabel"
          htmlFor={labelFor}
        >
          {label}
        </div>
        <div className="formValue">
          {children}
        </div>
      </div>
    );
  }
}

export default Formrow;
