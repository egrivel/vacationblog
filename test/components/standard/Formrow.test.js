import React from 'react';
import renderer from 'react-test-renderer';

import Formrow from '../../../src/components/standard/Formrow';

/*
 * Test strategy
 * The component has three optional attributes.
 * Confirm that the component renders correctly both with and without
 * the attributes.
 */
describe('components/standard/Formrow', () => {
  const testLabel = 'Test Label';
  const testLabelFor = 'test-label-for';
  const testContent = <span>This is some <em>test</em> content.</span>;

  describe('#render', () => {
    test('renders with label and value', () => {
      const render = renderer.create(
        <Formrow
          label={testLabel}
          labelFor={testLabelFor}
        >
          {testContent}
        </Formrow>
      );
      expect(render.toJSON()).toMatchSnapshot();
    });

    test('renders with label but no value', () => {
      const render = renderer.create(
        <Formrow />
      );
      expect(render.toJSON()).toMatchSnapshot();
    });
  });
});
