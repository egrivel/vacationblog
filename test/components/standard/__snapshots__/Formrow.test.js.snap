// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/standard/Formrow #render renders with label and value 1`] = `
<div>
  <div
    className="formLabel"
    htmlFor="test-label-for"
  >
    Test Label
  </div>
  <div
    className="formValue"
  >
    <span>
      This is some 
      <em>
        test
      </em>
       content.
    </span>
  </div>
</div>
`;

exports[`components/standard/Formrow #render renders with label but no value 1`] = `
<div>
  <div
    className="formLabel"
  />
  <div
    className="formValue"
  />
</div>
`;
